stellar-contract-env-host/src/host.rs

```diff
@@ -1441,8 +1488,54 @@ impl CheckedEnv for Host {
         i: RawVal,
         j: RawVal,
         l: RawVal,
-    ) -> Result<RawVal, HostError> {
-        todo!()
+    ) -> Result<Object, HostError> {
+        // offset in the host binary
+        let i: usize = u32::try_from(i).map_err(|_| {
+            HostError::WithStatus(
+                String::from("i must be u32"),
+                ScStatus::HostFunctionError(ScHostFnErrorCode::InputArgsWrongType),
+            )
+        })? as usize;
+        // offset in the guest memory
+        let j: u32 = j.try_into().map_err(|_| {
+            HostError::WithStatus(
+                String::from("j must be u32"),
+                ScStatus::HostFunctionError(ScHostFnErrorCode::InputArgsWrongType),
+            )
+        })?;
+        // length of the slice to copy
+        let l: usize = u32::try_from(l).map_err(|_| {
+            HostError::WithStatus(
+                String::from("l must be u32"),
+                ScStatus::HostFunctionError(ScHostFnErrorCode::InputArgsWrongType),
+            )
+        })? as usize;
+
+        let res = self.visit_obj(b, move |hv: &Vec<u8>| {
+            if i > u32::MAX as usize - l {
+                return Err(HostError::WithStatus(
+                    String::from("u32 overflow"),
+                    ScStatus::HostFunctionError(ScHostFnErrorCode::InputArgsInvalid),
+                ));
+            }
+            let vnew: Vec<u8> = self.with_current_frame(|frame| match frame {
+                #[cfg(feature = "vm")]
+                Frame::ContractVM(vm) => vm.with_memory_access(|mem| {
+                    let mut vnew = hv.clone();
+                    /* we would potentially let the vec grow */
+                    if (i + l) > vnew.len() {
+                        vnew.resize(i + l, 0);
+                    }
+                    let _ = mem.get_into(j, &mut vnew.as_mut_slice()[i..(i + l)])?;
```

Comment:
Nitpick: why do you need `let _ =` here?

Reply:
Fixed.